core/log/tau.cpp

```diff
@@ -30,16 +30,14 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ******************************<GINKGO LICENSE>*******************************/
 
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/log/profiler_hook.hpp>
```

Comment:
```suggestion
#include <ginkgo/core/base/exception_helpers.hpp>
#include <ginkgo/core/log/profiler_hook.hpp>
#include <ginkgo/config.hpp>
```
nit: because it uses GKO_HAVE_TAU

Reply:
ah. this order fits the ginkgo include order more.
profiler_hook.hpp should be the main header which includes the config.
```
#include <ginkgo/core/log/profiler_hook.hpp>


#if GKO_HAVE_TAU
#define PERFSTUBS_USE_TIMERS
#include <perfstubs_api/timer.h>
#endif


#include <ginkgo/core/base/exception_helpers.hpp>
```
@tcojean do you think we still need to include the config after exception_herlpers?